src/devices/Common/Iot/Device/Graphics/BitmapImage.cs

```diff
@@ -49,12 +56,82 @@ protected BitmapImage(byte[] data, int width, int height, int stride)
         public int Stride { get; }
 
         /// <summary>
-        /// Sets pixel at specific position
+        /// The format of the image
         /// </summary>
-        /// <param name="x">X coordinate of the pixel</param>
-        /// <param name="y">Y coordinate of the pixel</param>
-        /// <param name="color">Color to set the pixel to</param>
-        public abstract void SetPixel(int x, int y, Color color);
+        public PixelFormat PixelFormat { get; }
+
+        /// <summary>
+        /// Accesses the pixel at the given position
+        /// </summary>
+        /// <param name="x">Pixel X position</param>
+        /// <param name="y">Pixel Y position</param>
+        public Color this[int x, int y]
+        {
+            get
+            {
+                return GetPixel(x, y);
+            }
+            set
+            {
+                SetPixel(x, y, value);
+            }
+        }
+
+        /// <summary>
+        /// Register an image factory.
+        /// </summary>
+        /// <param name="factory">The image factory to register</param>
+        public static void RegisterImageFactory(IImageFactory factory)
+        {
+            s_currentFactory = factory;
+        }
+
+        /// <summary>
+        /// Creates a bitmap using the active factory. This requires an implementation to be registered.
+        /// See <see cref="BitmapImage"/> for details.
+        /// </summary>
+        /// <param name="width">Width of the image, in pixels</param>
+        /// <param name="height">Height of the image, in pixels</param>
+        /// <param name="pixelFormat">Desired pixel format</param>
+        /// <returns>A new bitmap with the provided size</returns>
+        public static BitmapImage CreateBitmap(int width, int height, PixelFormat pixelFormat)
+        {
+            VerifyFactoryAvailable();
+            return s_currentFactory!.CreateBitmap(width, height, pixelFormat);
+        }
+
+        private static void VerifyFactoryAvailable()
```

Comment:
if you put `[MemberNotNull(nameof(s_currentFactory))]` on this method then you won't need `s_currentFactory!` everywhere